Moderation Session docs update (#18143)
gravitational/teleport: docs/pages/access-controls/guides/moderated-sessions.mdx

|---|---|---|
|`on_leave`|[On leave](#on-leave)|The action to take when the policy is no longer satisfied|

#### Example Session Moderation Required Role

The policy below specifies that the `prod-access` role must have a minimum of
two users with the role `auditor` and the mode `moderator` present in the
session to start it. The policy applies to SSH and Kubernetes sessions only.

This policy requires that at least one user with the `auditor` role is present
as a moderator for SSH or Kubernetes sessions to start. That is applied for

> Servers and Kubernetes resources labeled `env: prod`. The session will not start

Reviewer: ```suggestion
servers and Kubernetes resources labeled `env: prod`. The session will not start
```
"server" is not a proper noun

until the policy is fulfilled.

```yaml
kind: role
metadata:
  name: prod-access

> version: v5

Reviewer: We should use v6 with the `kubernetes_resources` key

|---|---|---|
|`on_leave`|[On leave](#on-leave)|The action to take when the policy is no longer satisfied|

> #### Example Session Moderation Required Role

Reviewer 2: This reads like a proper noun, but we don't really have a first class resource called "Session Moderation Required Role".

I would personally just keep `Example` here.

Reviewer 3: Or the confusion here may be because we don't use title case in our titles, for whatever reason.

Reviewer: This is to make it clearer which terms in a heading are proper nouns

The policy below specifies that the `prod-access` role must have a minimum of
two users with the role `auditor` and the mode `moderator` present in the
session to start it. The policy applies to SSH and Kubernetes sessions only.

When a user with this require policy starts a session, it will be pending until

> the policy is fulfilled. The role matches to `env: prod` labeled `ssh` and `k8s`

Reviewer 2: nit: `ssh` and `k8s` aren't resources, they're session kinds.

resources. 

> 

Reviewer 2: Let's add a plain-text description of what this policy is doing. Something like:

> This policy requires that at least one user with the `auditor` role is present in order for SSH or Kubernetes sessions to begin.

```yaml
kind: role
metadata:
  name: prod-access

> version: v5

Reviewer 2: 🤦 great catch. this is critical!

|---|---|---|
|`on_leave`|[On leave](#on-leave)|The action to take when the policy is no longer satisfied|

#### Example Session Moderation Required Role

The policy below specifies that the `prod-access` role must have a minimum of
two users with the role `auditor` and the mode `moderator` present in the
session to start it. The policy applies to SSH and Kubernetes sessions only.

This policy requires that at least one user with the `auditor` role is present
as a moderator for SSH or Kubernetes sessions to start. That is applied for

> Servers and Kubernetes resources labeled `env: prod`. The session will pend starting

Reviewer 2: ```suggestion
Servers and Kubernetes resources labeled `env: prod`. The session will not start
```

|`kinds`|`[]`[Session kind](#session-kinds)|The kind of session that the policy applies to|
|`modes`|`[]`[Participant mode](#participant-modes)|The participant mode that applies to the user joining the Moderated Session under this policy|

<Admonition type="note">

> Moderator or observers can only join via CLI currently with `tsh join --mode=observer|moderator <session id>`.  Web joining is 

Reviewer 2: I would rephrase this a bit.

> In order to join a session in `moderator` or `observer` mode, users must join
> using the CLI: `tsh join --mode=observer|moderator`.
> Joining from the web UI always joins in `participant` mode.

Author: Modified to mention k8s not available from UI as this is now available for Servers

|`kinds`|`[]`[Session kind](#session-kinds)|The kind of session that the policy applies to|
|`modes`|`[]`[Participant mode](#participant-modes)|The participant mode that applies to the user joining the Moderated Session under this policy|

<Admonition type="note">

> Joining from the UI is available for Server for SSH sessions.  Kubernetes are only joinable by CLI currently.

Reviewer 2: ```suggestion
Joining from the UI is available for SSH sessions.  Kubernetes sessions can only be joined from the CLI.
```

kind: role
metadata:
  name: auditor

> version: v5

Reviewer: We should use v6

Author: updated @ptgott

        modes: ['moderator', 'observer']
```

#### Joining sessions example

Here is an example of Jeff with role `prod-access` connecting to
a SSH server in the production environment.

```code
$ tsh ssh ubuntu@prod.teleport.example.com
Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
Teleport > Controls
  - CTRL-C: Leave the session
  - t: Forcefully terminate the session (moderators only)
Teleport > User jeff joined the session.
Teleport > Waiting for required participants...
```

Jeff's session is paused, waiting for the required observers.

Now Alice with the `auditor` role joins as a moderator and
the session can begin. 

```code
$ tsh join --mode=moderator 46e2af03-62d6-4e07-a886-43fe741ca044
Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
Teleport > Controls
  - CTRL-C: Leave the session
  - t: Forcefully terminate the session (moderators only)
Teleport > User jeff joined the session.
Teleport > Waiting for required participants...
Teleport > User alice joined the session.
Teleport > Connecting to prod.teleport.example.com over SSH

ubuntu@prod.teleport.example.com % 
```

> Here is an example of joining from the UI that is available for Server sessions.

Reviewer: ```suggestion
Here is an example of joining from the UI that is available for server sessions.
```

![Join Server Session from UI](../../../img/webui-active-session.png)

<Admonition title="Note" type="tip">

> Joining as a moderator and observer is available for Servers in the UI Console.

Reviewer: - "Servers" should be "servers", since it's not a proper noun
- What is the UI Console?

Author: This meant Web UI but it's redundant since we're showing the joining of servers.  Removed

        modes: ['moderator', 'observer']
```

#### Joining sessions example

> Here is an example of Jeff with role `prod-access` joining to a 
> production environment.  

Reviewer 2: ```suggestion
Here is an example of Jeff with role `prod-access` connecting to
a SSH server in the production environment.
```

```code
$ tsh ssh ubuntu@prod.teleport.example.com
Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
Teleport > Controls
  - CTRL-C: Leave the session
  - t: Forcefully terminate the session (moderators only)
Teleport > User jeff joined the session.
Teleport > Waiting for required participants...
```

> Now Alice with the `auditor` role joins as a moderator and
> the session can begin. 

Reviewer 2: ```suggestion
Jeff's session is paused, waiting for the required observers.

Now Alice with the `auditor` role joins as a moderator and
the session can begin. 
```

```

#### Joining sessions example

Here is an example of Jeff with role `prod-access` connecting to
a SSH server in the production environment.

```code
$ tsh ssh ubuntu@prod.teleport.example.com
Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
Teleport > Controls
  - CTRL-C: Leave the session
  - t: Forcefully terminate the session (moderators only)
Teleport > User jeff joined the session.
Teleport > Waiting for required participants...
```

Jeff's session is paused, waiting for the required observers.

Now Alice with the `auditor` role joins as a moderator and
the session can begin. 

```code
$ tsh join --mode=moderator 46e2af03-62d6-4e07-a886-43fe741ca044
Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
Teleport > Controls
  - CTRL-C: Leave the session
  - t: Forcefully terminate the session (moderators only)
Teleport > User jeff joined the session.
Teleport > Waiting for required participants...
Teleport > User alice joined the session.
Teleport > Connecting to prod.teleport.example.com over SSH

ubuntu@prod.teleport.example.com % 
```

Here is an example of joining from the UI that is available for Server sessions.

> ![Join Server Session from UI](../../../img/access-controls/joinsession.png)

Reviewer 2: We're about to change this, so I would either remove this section if you want to merge this now, or wait 1-2 weeks for an updated screenshot.

Author: Thanks let's wait then
